### ISSUE TITLE
Error encoding type where underlying value is slice of built-in type

### ISSUE THREAD

**Issue by klauspost** [pquerna/ffjson#44]
Encoding this gives multiple errors in the generated code:

```Go
package test

import "encoding/xml"

type Custom struct {
	Header *xml.CharData `json:",omitempty"`
}
```

```.\test_ffjson.go:37: cannot range over mj.Header (type *xml.CharData)```

The generated code:
```Go
				for i, v := range mj.Header {
					if i != 0 {
						buf.WriteString(`,`)
					}
					fflib.FormatBits(&scratch, buf, uint64(v), 10, false)
				}
```
Here "range mj.Header" should be "range *mj.Header". Should be fixable.

However, the errors in the decoder:
```
.\test_ffjson.go:184: cannot use make([]uint8, 0) (type []uint8) as type *xml.CharData in assignment
.\test_ffjson.go:238: first argument to append must be slice; have *xml.CharData
```
These seem a bit harder to fix, since a fix would require importing "encoding/xml", or maybe just fall back on this type of values.

**Comment by pquerna** [comment]
this specific case is partially fixed by 9cf92d1

**Comment by klauspost** [comment]
To me this seems fixed At least I don't get invalid code anymore. Should this be closed?
